crates/iceberg/src/delete_file_index.rs

```diff
@@ -193,26 +199,3 @@ impl PopulatedDeleteFileIndex {
         results
     }
 }
-
-/// Future for the `DeleteFileIndex::get_deletes_for_data_file` method
-pub(crate) struct DeletesForDataFile<'a> {
-    state: Arc<RwLock<DeleteFileIndexState>>,
-    data_file: &'a DataFile,
-    seq_num: Option<i64>,
-}
-
-impl Future for DeletesForDataFile<'_> {
-    type Output = Result<Vec<FileScanTaskDeleteFile>>;
-
-    fn poll(self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<Self::Output> {
-        match self.state.try_read() {
-            Ok(guard) => match guard.deref() {
-                DeleteFileIndexState::Populated(idx) => Poll::Ready(Ok(
-                    idx.get_deletes_for_data_file(self.data_file, self.seq_num)
-                )),
-                _ => Poll::Pending,
-            },
-            Err(err) => Poll::Ready(Err(Error::new(ErrorKind::Unexpected, err.to_string()))),
```

Comment:
We have 2 possible problems here:
1. `try_read` returns `TryLockError::WouldBlock`, which should not be an error, but `Pending`.
2. More severely, we don't have any `Waker` set, so once `Pending`, the future will sleep forever.

Therefore, I think it's better to replace manual `Future` implementation with other sync primitives.

Reply:
I've addressed the missing Waker in my open PR in my open PR: https://github.com/apache/iceberg-rust/pull/982/commits/5f0b07381bb66db8e74d7285a4bfaace040bfd80

Reply:
which PR？

BTW, it seems in this commit the TryLock problem is not resolved

Reply:
BTW, I think manual `Future` implementation should be discouraged in favor of `async fn` with synchronisation primitives.

Reply:
Oh, it's in #982